python/lsst/rubintv/production/plotting/nightReportPlotBase.py

```diff
@@ -0,0 +1,282 @@
+
+    @abstractmethod
+    def createAndUpload(self, *args):
+        """Create the plot defined in ``plot`` and upload it.
+
+        This is the method called by the Night Report channel to create the
+        plot and send it to the bucket.
+
+        Parameters
+        ----------
+        *arg : `any`
+            The arguments which are passed to the `plot` function.
+        """
+        raise NotImplementedError()
+
+
+class LatissPlot(BasePlot):
+    """Base class for LATISS night report plots.
+
+    Parameters
+    ----------
+    dayObs : `int`
+        The dayObs to make the plots for.
+    plotName : `str`
+        The name of the plot, used for upload.
+    plotGroup : `str`
+        The group to put the plot in on the front end.
+    locationConfig : `lsst.rubintv.production.utils.LocationConfig`, optional
+        The locationConfig containing the paths, or ``None`` if being used for
+        development
+    uploader : `lsst.rubintv.production.Uploader`, optional
+        The uploader, or ``None``, if being used for development.
+    """
+    def __init__(self, *,
+                 dayObs,
+                 plotName,
+                 plotGroup,
+                 locationConfig,
+                 uploader):
+
+        super().__init__(dayObs=dayObs,
+                         plotName=plotName,
+                         plotGroup=plotGroup,
+                         channelName="auxtel_night_reports",
+                         locationConfig=locationConfig,
+                         uploader=uploader,
+                         )
+
+    @abstractmethod
+    def plot(self, nightReport, metadata, ccdVisitTable):
+        """Subclasses must implement this method.
+
+        These are the parameters passed automatically when making LATISS plots.
+
+        Parameters
+        ----------
+        nightReport : `lsst.rubintv.production.nightReport.NightReport`
+            The night report for the current night.
+        metadata : `pandas.DataFrame`
+            The front page metadata, as a dataframe.
+        ccdVisitTable : `pandas.DataFrame`
+            The visit summary table for the current day.
+
+        Returns
+        -------
+        success : `bool`
+            Did the plotting succeed, and thus upload should be performed?
+        """
+        raise NotImplementedError()
+
+    def createAndUpload(self, nightReport, metadata, ccdVisitTable):
+        """Create the plot defined in ``plot`` and upload it.
+
+        This is the method called by the Night Report channel to create the
+        plot and send it to the bucket.
+
+        Parameters
+        ----------
+        nightReport : `lsst.rubintv.production.nightReport.NightReport`
+            The night report for the current night.
+        metadata : `pandas.DataFrame`
+            The front page metadata, as a dataframe.
+        ccdVisitTable : `pandas.DataFrame`
+            The visit summary table for the current day.
+        """
```

Comment:
No need to repeat docs from the base class if they're the same; they'll get inherited if there is no docstring.  Lots of cases where this applies through this PR.

Reply:
Thanks - deleted.